airline-core/src/main/java/com/github/rvesse/airline/HelpOption.java

```diff
@@ -19,7 +19,7 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import javax.inject.Inject;
+import jakarta.inject.Inject;
```

Comment:
Doing that means that we will be forced to have the Jakarta ones around and in most cases, you definitely don't want to have both in parallel because it's going to make your life hell (your CDI implementation will support only one of them and your IDE will keep suggesting either of them, which will lead to errors).

I think in the core artifact, you shouldn't depend on these annotations and use another mechanism (or an internal annotation, preferably with a name that is a bit different such as `AirlineInject`) to inject things.

Reply:
Yes my intent is to have a new Airline specific annotation as the default, likely `@Module`, just didn't get that far with this OR yet (hence why it's still draft)

One of my main concerns is not breaking everyone when they upgrade so I wanted to make sure I had that working first

Reply:
Note as long as it's possible to exclude the Jakarta dependency with Maven and that everything still works, it's fine by me i.e. you could autodetect if and which `@Inject` is around.

Reply:
- [ ] Verify excluding the injection dependencies allows normal functionality

Reply:
Verified with the `airline-examples` module that things work fine when excluding the two `@Inject` dependencies